modules/ssl/ssl_private.h

```diff
@@ -1150,10 +1150,12 @@ void ssl_init_ocsp_certificates(server_rec *s, modssl_ctx_t *mctx);
 
 #endif
 
+#ifndef MODSSL_USE_OPENSSL_PRE_1_1_API
```

Comment:
Shouldn't this be `#if MODSSL_USE_OPENSSL_PRE_1_1_API`?
BTW: I think there are cases where `MODSSL_USE_OPENSSL_PRE_1_1_API` is not defined at all, in case of `LIBRESSL_VERSION_NUMBER < 0x2060000f` https://github.com/apache/httpd/blob/0dbc5ca0769aa35f0b956294b2c85726d3ea6466/modules/ssl/ssl_private.h#L131-L149
Wouldn't that be an issue for all the `#if MODSSL_USE_OPENSSL_PRE_1_1_API` code?
```suggestion
#if MODSSL_USE_OPENSSL_PRE_1_1_API
```

Reply:
Thanks @rpluem, I pushed a commit to fix that first point.  w.r.t. second point, I have no idea, if it is broken somebody with LibreSSL will have to show us how to fix it.

Reply:
Adding MODSSL_USE_OPENSSL_PRE_1_1_API define in the LIBRESSL_VERSION < 0x2060000f case makes it build with libressl-2.5.5. The code builds with Libressl 3.3.3 as well.
```
diff --git a/modules/ssl/ssl_private.h b/modules/ssl/ssl_private.h
index 7eb9a364b0..5ba968be41 100644
--- a/modules/ssl/ssl_private.h
+++ b/modules/ssl/ssl_private.h
@@ -137,6 +137,7 @@
         SSL_CTX_ctrl(ctx, SSL_CTRL_SET_MIN_PROTO_VERSION, version, NULL)
 #define SSL_CTX_set_max_proto_version(ctx, version) \
         SSL_CTX_ctrl(ctx, SSL_CTRL_SET_MAX_PROTO_VERSION, version, NULL)
+#define MODSSL_USE_OPENSSL_PRE_1_1_API (1)
 #elif LIBRESSL_VERSION_NUMBER < 0x2070000f
 /* LibreSSL before 2.7 declares OPENSSL_VERSION_NUMBER == 2.0 but does not
  * include most changes from OpenSSL >= 1.1 (new functions, macros,
```

Reply:
Possibly something like this then?
```
Index: modules/ssl/ssl_private.h
===================================================================
--- modules/ssl/ssl_private.h	(revision 1889960)
+++ modules/ssl/ssl_private.h	(working copy)
@@ -142,8 +142,8 @@
  * include most changes from OpenSSL >= 1.1 (new functions, macros, 
  * deprecations, ...), so we have to work around this...
  */
-#define MODSSL_USE_OPENSSL_PRE_1_1_API (1)
 #endif /* LIBRESSL_VERSION_NUMBER < 0x2060000f */
+#define MODSSL_USE_OPENSSL_PRE_1_1_API (LIBRESSL_VERSION_NUMBER < 0x2070000f)
 #else /* defined(LIBRESSL_VERSION_NUMBER) */
 #define MODSSL_USE_OPENSSL_PRE_1_1_API (OPENSSL_VERSION_NUMBER < 0x10100000L)
 #endif
```

Reply:
Looks sensible

Reply:
Merged except I have not tried to incorporate the Libre changes since I can't test them, FYI @bigio